forge-game/src/main/java/forge/game/GameActionUtil.java

```diff
@@ -248,6 +248,14 @@ public static List<SpellAbility> getAlternativeCosts(final SpellAbility sa, fina
                     alternatives.add(foretold);
                 }
 
+                if (source.isWarped() && activator.equals(source.getOwner()) && !source.enteredThisTurn() && !source.getManaCost().isNoCost()) {
+                    final SpellAbility normalWarp = sa.copy(activator);
```

Comment:
you need to build it earlier so real AltCost see it here
otherwise how can friend Henzie blitz it?

Reply:
I don't know
casting from Warp isn't an Alternative Cost (but Alternative Zone)

so my Gut says, you can Blitz warped Cards from Exile

Reply:
I will do Blitz stuff later